README.md

```diff
@@ -23,110 +23,131 @@ $ composer require tomphp/config-service-provider
 ```php
 <?php
 
-use League\Container\Container;
-use League\Container\ServiceProvider\AbstractServiceProvider;
-use TomPHP\ConfigServiceProvider\ConfigServiceProvider;
-
-class DatabaseConnectionProvider extends AbstractServiceProvider
-{
-    protected $provides = [
-        'database_connection',
-    ];
-    
-    public function register()
-    {
-        $this->container->share('database_connection', function () {
-            return new DatabaseConnection(
-                $this->container->get('config.db.name'),
-                $this->container->get('config.db.username'),
-                $this->container->get('config.db.password')
-            );
-        });
-    }
-}
+use League\Container\Container; // or Pimple\Container
+use TomPHP\ConfigServiceProvider\ConfigureContainer;
 
-$appConfig = [
+$config = [
     'db' => [
         'name'     => 'example_db',
         'username' => 'dbuser',
         'password' => 'dbpass',
-    ]
+    ],
+
+    'di' => [
+        'services' => [
+            'database_connection' => [
+                'class' => DatabaseConnection::class,
+                'arguments' => [
+                    'config.db.name',
+                    'config.db.username',
+                    'config.db.password',
+                ],
+            ],
+        ],
+    ],
 ];
 
 $container = new Container();
-
-$container->addServiceProvider(ConfigServiceProvider::fromConfig($appConfig));
-$container->addServiceProvider(new DatabaseConnectionProvider());
+ConfigureContainer::fromArray($container, $config);
```

Comment:
@tomphp 

I have been thinking about the naming, and I am not entirely sure, but maybe we can come up with something better, what do you think?

Just wondering, maybe a usage similar to this makes sense:

```php
ContainerConfigurator::for($container)
    ->usingSettings($settings)
    ->usingConfiguration($configuration)
    ->usingConfigurationFiles($files)
    ->apply()
;
```
Is that too much?

A bit like a builder, instead, it doesn't return anything.